src/defense.cpp

```diff
@@ -1154,6 +1161,14 @@ std::vector<itype_id> caravan_items(caravan_category cat)
                   "ak47", "762_m87", "m4a1", "556", "savage_111f", "hk_g3",
                   "762_51", "hk_g80", "12mm", "plasma_rifle", "plasma"
               };
+
+        // Add the default magazine types for each gun.
+        for (unsigned i = 0; i < ret.size(); i++) {
```

Comment:
@mugling Hmm, I wanted to test out whether range-based loops behave poorly if you manipulate the structure's contents while you're iterating (I recall such problems with C++03 std::foo::iterator objects when removing elements). I assumed that since I'm only *appending* to it, things should be sane, right? Well... range-based loops are now confusing me:

https://gist.github.com/tycho/fe660932f9ff44206c79d04174f7aaa9#file-range-loop-test-L37-L38

Note how lines 37 and 38 are identical. Did it iterate over the same element twice and skip one?

If I replace the middle range loop with an index-based one, it behaves as you'd expect.